### PR TITLE
fix(service-worker): remove `controllerchange` listener when app is destroyed

### PR DESCRIPTION
**Pull request by arturovt**
This commit updates the `ngswAppInitializer` implementation and removes the `controllerchange` listener upon the destruction of the `ApplicationRef`. This adjustment aims to prevent memory leaks. In a zone.js environment, neglecting to do so could lead to the perpetual creation of a zone task, which captures the zone and obstructs proper garbage collection.

### CODE REVIEW

File: packages/service-worker/src/provider.ts [hunk 1]
```diff
@@ -48,10 +49,13 @@ export function ngswAppInitializer(
...
-        if (navigator.serviceWorker.controller !== null) {
-          navigator.serviceWorker.controller.postMessage({action: 'INITIALIZE'});
-        }
+      const onControllerChange = () =>
+        navigator.serviceWorker.controller?.postMessage({action: 'INITIALIZE'});
+
+      navigator.serviceWorker.addEventListener('controllerchange', onControllerChange);
+
+      appRef.onDestroy(() => {
+        navigator.serviceWorker.removeEventListener('controllerchange', onControllerChange);
```
Comment (AndrewKushnir) [inline packages/service-worker/src/provider.ts:58]: nit: for better minification, we can extract the `navigator.serviceWorker` to a separate const:
```suggestion
      const sw = navigator.serviceWorker;
      const onControllerChange = () =>
        sw.controller?.postMessage({action: 'INITIALIZE'});

      sw.addEventListener('controllerchange', onControllerChange);

      appRef.onDestroy(() => {
        sw.removeEventListener('controllerchange', onControllerChange);
```

**Review by AndrewKushnir — COMMENTED** [review]
@arturovt thanks for the fix! The code looks great, just added 1 minor comment.

File: packages/service-worker/src/provider.ts (hunk 1 above)
Comment (arturovt) [reply packages/service-worker/src/provider.ts]: Updated.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
**Caretaker note**: this change is low-risk, no additional reviews are required, this PR is ready for merge.

**Pull request closed without merging**